ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -39,6 +39,13 @@ azure_connection::azure_connection(const std::string &bucket_name, const std::st
         std::getenv("AZURE_STORAGE_CONNECTION_STRING"), bucket_name)),
       _bucket_name(bucket_name), _object_prefix(obj_prefix)
 {
+    // Confirm that we can access the bucket, else fail.
+    bool exists;
+    int ret = bucket_exists(exists);
+    if (!exists)
+        std::cerr << _bucket_name + " : No such bucket." << std::endl;
```

Comment:
I think we should return a -1 on top of the error message

Reply:
agreed :)

Reply:
missing brackets around the if statements :)

Reply:
reverted - can't return with value in [constructor](https://www.google.com/search?q=return+-1+in+constructor+cpp&rlz=1C5GCEM_enAU1035AU1036&ei=DxTOY9XLA6aa4-EPyp2KmAs&ved=0ahUKEwiVj8uM9Nz8AhUmzTgGHcqOArMQ4dUDCA8&uact=5&oq=return+-1+in+constructor+cpp&gs_lcp=Cgxnd3Mtd2l6LXNlcnAQAzoKCAAQRxDWBBCwA0oECEEYAEoECEYYAFC3DVi3DWCmD2gEcAB4AIABsgGIAbIBkgEDMC4xmAEAoAEByAEIwAEB&sclient=gws-wiz-serp)

Reply:
closing this due to impl of exceptions